### ISSUE TITLE
Drawer - automatically focus on first interactable element when drawer is opened

### ISSUE THREAD

**Issue by nicolethoen** [patternfly/patternfly-react#8534]
Follow up for core issue https://github.com/patternfly/patternfly/issues/5319
A more consistent solution than was used to fix https://github.com/patternfly/patternfly-react/issues/8394

The modal component automatically places focus on the first interactable element in the modal when it is opened. 
In the case of the Drawer, that should be the close button by default. 
If the close button is not provided, then it should be the first interactable element in the drawer.
If there are no interactable elements, focus should be placed on the drawer body.

Like the modal, this should happen automatically for the drawer, unless overridden by the consumer.
